handling-forms/controlled-forms.md

```diff
@@ -637,24 +713,126 @@ export default NewStudentForm;
 
 ### !end-callout
 
+### !callout-info
+
+## Refactoring Opportunity: Combining Multiple Change Handlers into One
+
+Notice that the functions `onNameChange` and `onEmailChange` are nearly identical. The only difference between the two event handlers is that one updates the value for the name property and one updates the value for the email property in the formFields state object.
+
+<br/>
+
+We can refactor our code to combine  `onNameChange` and `onEmailChange` into one event handler called `handleChange` to reduce repetition in the `NewStudentForm` component.
+
+<br/>
+
+We can accomplish this by using the spread operator notation with object shorthand notation as we saw in the lesson on "Passing Down Event Handlers".
```

Comment:
This callout directly follows the callout above and visually takes up a lot of space. Maybe I should put the code in a detail to make this shorter

Reply:
Yeah, for the other ones I had done, the change wasn't relying on such new concepts, so I could give the reader a "think about how to do x" and then provide the solution collapsed. I think here, you're going to need to hint at the overall apporach in the main callout, then have the detailed explanation and code in the detail. Your currently summary (through "We can refactor our code to combine...") sounds good, but I'd like it to mention that the state key to be updated will come from the event data. Added a suggestion above.